src/test/java/dal/fatih/todoly/TodolyTest.java

```diff
@@ -1,252 +1,553 @@
+        );
+
+        ResponseEntity<String> responseEntity = this.testRestTemplate
+                .postForEntity(taskCreateUrl, request, String.class);
+
+        String actual = responseEntity.getBody();
+        String expected = "Due date must be a future date";
+        logger.warn(responseEntity.getBody());
+
+        assertThat(taskRepository.list().size(), is(equalTo(0)));
+        assertThat(responseEntity.getStatusCode(), equalTo(HttpStatus.BAD_REQUEST));
+        assertThat(actual, is(containsString(expected)));
     }
 
     @Test
-    public void shouldListAllTasks() {
-        addTask("title-of-the-task-to-be-listed", "description-of-task", "2030-05-05");
-        addTask("title-of-the-task-to-be-listed-2", "description-of-task", "2030-05-05");
-        provideInput(Arrays.asList("2", "q"));
-        App.main(new String[]{});
-        Assert.assertTrue(outContent.toString().contains("title-of-the-task-to-be-listed"));
-        Assert.assertTrue(outContent.toString().contains("title-of-the-task-to-be-listed-2"));
+    @DirtiesContext(methodMode = DirtiesContext.MethodMode.BEFORE_METHOD)
+    public void shouldNotAllowDueDateToBeEqualToNow() throws URISyntaxException {
+        URI taskCreateUrl = new URI(createURLWithPort("/task"));
+        HttpEntity<TaskDTO> request = new HttpEntity<TaskDTO>(
+                new TaskDTO("Title-of-task", "Description-of-task"
+                        , LocalDateTime.now())
+        );
+
+        ResponseEntity<String> responseEntity = this.testRestTemplate
+                .postForEntity(taskCreateUrl, request, String.class);
+
+        String actual = responseEntity.getBody();
+        String expected = "Due date must be a future date";
+        logger.warn(responseEntity.getBody());
+
+        assertThat(taskRepository.list().size(), is(equalTo(0)));
+        assertThat(responseEntity.getStatusCode(), equalTo(HttpStatus.BAD_REQUEST));
+        assertThat(actual, is(containsString(expected)));
     }
 
     @Test
-    public void shouldFindNoTaskToShowDetails() {
-        provideInput(Arrays.asList("3", "46856845672662", "q"));
-        App.main(new String[]{});
-        Assert.assertTrue(outContent.toString().contains("No task found"));
+    @DirtiesContext(methodMode = DirtiesContext.MethodMode.BEFORE_METHOD)
+    public void shouldListAllTasks() throws URISyntaxException {
+        LocalDateTime dueDate = LocalDateTime.now().plusMinutes(10);
+        String title = "Listed-title-of-task", description = "Listed-description-of-task";
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd-MM-yyyy HH:mm:ss");
+
+        createTask(title, description, dueDate, null, null, null);
+
+        URI urlOfGetAll = new URI(createURLWithPort("/tasks"));
+        ResponseEntity<String> responseEntity =
+                this.testRestTemplate.getForEntity(urlOfGetAll, String.class);
+        logger.warn(responseEntity.getBody());
+
+        String actual = responseEntity.getBody();

```

Comment:
Parse the string body as json into `HashMap` by using `ObjectMapper` from `fasterxml`